FirebaseAdmin/FirebaseAdmin/Auth/Hash/RepeatableHash.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+
+namespace FirebaseAdmin.Auth.Hash
+{
+    /// <summary>
+    /// An abstract <a cref="UserImportHash">UserImportHash</a> implementation for specifying a
+    /// <c>Rounds</c> count in a given range.
+    /// </summary>
+    public abstract class RepeatableHash : UserImportHash
+    {
+        private int? rounds;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RepeatableHash"/> class.
+        /// Propogates the name to UserImportHash.
+        /// </summary>
+        /// <param name="hashName">The name of the hashing algorithm.</param>
+        public RepeatableHash(string hashName)
+            : base(hashName) { }
+
+        /// <summary>
+        /// Gets or sets the number of rounds for the repeatable hash. Verifies that the
+        /// specified <c>Rounds</c> are within the required bounds.
+        /// </summary>
+        public int Rounds
```

Comment:
Keep this as a regular getter/setter without validation. Perform all validation in `GetOptions()`. Here and in other similar classes.